src/main.c

```diff
@@ -257,63 +257,52 @@ int main(int argc, char* args[])
   {
     char* defaultEggName = "game.egg";
     char* mainFileName = "main.wren";
-    char* fileName = defaultEggName;
 
     char* base = BASEPATH_get();
     char* arg = optparse_arg(&options);
-    if (arg != NULL) {
-      fileName = arg;
-    } else {
-      fileName = defaultEggName;
-    }
 
-    size_t baseLen = strlen(base);
-    size_t fileNameLen = strlen(fileName);
-    char* pathBuf = calloc(baseLen + fileNameLen + 1, sizeof(char));
-    strcpy(pathBuf, base);
-    strcat(pathBuf, fileName);
-    // Re-target basepath to the argument given
-    if (isDirectory(pathBuf)) {
-      BASEPATH_set(pathBuf);
-      fileName = defaultEggName;
-      fileNameLen = strlen(fileName);
-      arg = NULL;
-    } else {
-      BASEPATH_set(dirname(pathBuf));
-    }
-    base = BASEPATH_get();
-    baseLen = strlen(base);
+    char pathBuf[PATH_MAX];
+
+    char* fileName = NULL;
+
+    if (arg != NULL) {
+      strcpy(pathBuf, arg);
```

Comment:
This needs to be strcat, and we need a `strcpy(pathBuf, base)` here, because otherwise on some OS, a relative file argument like `../` will fail.

Reply:
Updated. Can you verify it works on debian now?

Reply:
So this works to solve the issue if you run `./dome ../`, but it doesn't work for `./dome ../game.egg`. I'm trying to pin down the root cause.